dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -37,49 +37,96 @@ public class TextMemorySkill
     /// </summary>
     public const string KeyParam = "key";
 
+    /// <summary>
+    /// Name of the context variable used to specify the number of memories to recall
+    /// </summary>
+    public const string LimitParam = "limit";
+
     private const string DefaultCollection = "generic";
     private const string DefaultRelevance = "0.75";
+    private const string DefaultLimit = "1";
 
     /// <summary>
-    /// Recall a fact from the long term memory
+    /// Key-based lookup for a specific memory
+    /// </summary>
+    /// <example>
+    /// SKContext[TextMemorySkill.KeyParam] = "countryInfo1"
+    /// {{memory.retrieve }}
+    /// </example>
+    /// <param name="context">Contains the 'collection' containing the memory to retrieve and the `key` associated with it.</param>
+    [SKFunction("Key-based lookup for a specific memory")]
+    [SKFunctionName("Retrieve")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection associated with the memory to retrieve",
+        DefaultValue = DefaultCollection)]
+    [SKFunctionContextParameter(Name = KeyParam, Description = "The key associated with the memory to retrieve")]
+    public async Task<string> RetrieveAsync(SKContext context)
+    {
+        var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
+        Verify.NotEmpty(collection, "Memory collection not defined");
+
+        var key = context.Variables.ContainsKey(KeyParam) ? context[KeyParam] : string.Empty;
+        Verify.NotEmpty(key, "Memory key not defined");
+
+        context.Log.LogTrace("Recalling memory from collection '{0}'", collection);
+
+        var memory = await context.Memory.GetAsync(collection, key);
+        return memory != null ? memory.Text : string.Empty;
```

Comment:
Another nit: wouldn't this be more succint:

return memory?.Text ?? string.Empty;

Reply:
Done